container/container.go

```diff
@@ -84,7 +85,11 @@ type Container struct {
 	LogPath         string
 	Name            string
 	Driver          string
-	OS              string
+
+	// Deprecated: use [ImagePlatform.OS] instead
+	OS string
```

Comment:
OK, this is a silly-question, because we have too many `Container` structs in our code, so I lost a bit which one's used for what (note to self: we should really improve GoDoc for these types to not only describe "what's in them", but also _what their purpose is_).

I _think_ this `Container` type is purely for storing state on-disk, correct? So preserving this field for backward-compatibility is backward-compatibility with existing state on-disk (up/downgrade dockerd), and this type is not returned over the API (so no compatibility issue there).

This PR adds a migration step during restore, so state should be migrated, which means that we could probably draw a roadmap when it's time to remove (1 .. 2 versions?). And it may be good to leave a comment in places where that's relevant.

I've ran into "this field is kept for backward compatibility" comments on various locations where it's no longer clear "... erm.. backward compatible with _what_? is this still relevant?", and such code never being removed because nobody knows 😂

Reply:
> I _think_ this `Container` type is purely for storing state on-disk, correct? So preserving this field for backward-compatibility is backward-compatibility with existing state on-disk (up/downgrade dockerd), and this type is not returned over the API (so no compatibility issue there).

Correct.

> This PR adds a migration step during restore, so state should be migrated, which means that we could probably draw a roadmap when it's time to remove (1 .. 2 versions?). And it may be good to leave a comment in places where that's relevant.

Good point, I'll open a tracking ticket for that.

Not sure if we can remove it that quickly though. If we remove it in eg. v30, the migration won't work for a pre-v28 daemons that skip the intermediate versions and update to v30+ right away.

Reply:
Opened https://github.com/moby/moby/issues/48892 and mentioned it in a TODO comment.